pkg/tsdb/elasticsearch/elasticsearch.go

```diff
@@ -2,50 +2,110 @@ package elasticsearch
+	})
+	if err := s.BackendPluginManager.Register("elasticsearch", factory); err != nil {
+		eslog.Error("Failed to register plugin", "error", err)
+	}
+	return nil
 }
 
-// New creates a new Executor func.
-// nolint:staticcheck // plugins.DataPlugin deprecated
-func New(httpClientProvider httpclient.Provider) func(*models.DataSource) (plugins.DataPlugin, error) {
-	// nolint:staticcheck // plugins.DataPlugin deprecated
-	return func(dsInfo *models.DataSource) (plugins.DataPlugin, error) {
-		return &Executor{
-			httpClientProvider: httpClientProvider,
-			intervalCalculator: interval.NewCalculator(),
-		}, nil
+// newService creates a new executor func.
+func newService(im instancemgmt.InstanceManager, httpClientProvider httpclient.Provider) *Service {
+	return &Service{
+		im:                 im,
+		HTTPClientProvider: httpClientProvider,
+		intervalCalculator: tsdb.NewCalculator(),
 	}
 }
 
-// Query handles an elasticsearch datasource request
-//nolint: staticcheck // plugins.DataResponse deprecated
-func (e *Executor) DataQuery(ctx context.Context, dsInfo *models.DataSource,
-	tsdbQuery plugins.DataQuery) (plugins.DataResponse, error) {
-	if len(tsdbQuery.Queries) == 0 {
-		return plugins.DataResponse{}, fmt.Errorf("query contains no queries")
+func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	if len(req.Queries) == 0 {
+		return &backend.QueryDataResponse{}, fmt.Errorf("query contains no queries")
 	}
 
-	client, err := es.NewClient(ctx, e.httpClientProvider, dsInfo, *tsdbQuery.TimeRange)
+	dsInfo, err := s.getDSInfo(req.PluginContext)
 	if err != nil {
-		return plugins.DataResponse{}, err
+		return &backend.QueryDataResponse{}, err
 	}
 
-	if tsdbQuery.Debug {
-		client.EnableDebug()
+	client, err := es.NewClient(ctx, s.HTTPClientProvider, dsInfo, req.Queries[0].TimeRange)
+	if err != nil {
+		return &backend.QueryDataResponse{}, err
 	}
 
-	query := newTimeSeriesQuery(client, tsdbQuery, e.intervalCalculator)
+	query := newTimeSeriesQuery(client, req.Queries, s.intervalCalculator)
 	return query.execute()
 }
+
+func newInstanceSettings() datasource.InstanceFactoryFunc {
+	return func(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		jsonData := map[string]interface{}{}
+		err := json.Unmarshal(settings.JSONData, &jsonData)
+		if err != nil {
+			return nil, fmt.Errorf("error reading settings: %w", err)
+		}
+		httpCliOpts, err := settings.HTTPClientOptions()
+		if err != nil {
+			return nil, fmt.Errorf("error getting http options: %w", err)
+		}
+		model := es.DatasourceInfo{
+			ID:             settings.ID,
+			URL:            settings.URL,
+			HTTPClientOpts: httpCliOpts,
+			Database:       settings.Database,
+			ESVersion:      jsonData["esVersion"].(string),
+			TimeField:      jsonData["timeField"].(string),
+			Interval:       jsonData["interval"].(string),
+			TimeInterval:   jsonData["timeInterval"].(string),
```

Comment:
Should check that these casts is allowed/ok. 

Regarding `esVersion` not sure that would handle version stored as either string or integer. Either need to move the version parsing that currently happens in client.go to to here and use a `*semver.Version` on the `DatasourceInfo ESVersion` field or either try cast to both string and integer.

Reply:
`esVersion` is being checked in `coerceVersion(v string)`?

Reply:
Was referring to something like this in `newInstanceSettings` since `jsonData["esVersion"]` could be either a string or an integer. 

```go
esVersion, err := coerceVersion(jsonData["esVersion"])
if err != nil {
	return nil, err
}

model.ESVersion = esVersion

func coerceVersion(v interface{}) (*semver.Version, error) {
	versionString, err := v.String()
	if err == nil {
		return semver.NewVersion(versionString)
	}

	versionNumber, err := v.Int()
	if err != nil {
		return nil, err
	}

	// Legacy version numbers (before Grafana 8)
	// valid values were 2,5,56,60,70
	switch versionNumber {
	case 2:
		return semver.NewVersion("2.0.0")
	case 5:
		return semver.NewVersion("5.0.0")
	case 56:
		return semver.NewVersion("5.6.0")
	case 60:
		return semver.NewVersion("6.0.0")
	case 70:
		return semver.NewVersion("7.0.0")
	default:
		return nil, fmt.Errorf("elasticsearch version=%d is not supported", versionNumber)
	}
}
```